compositor_render/src/renderer/transformation.rs

```diff
@@ -0,0 +1,22 @@
+use std::{collections::HashMap, error::Error};
+
+use compositor_common::scene::TransformationRegistryKey;
+
+use super::texture::Texture;
+
+#[derive(Debug)]
+pub enum TransformationParams {
+    String(String),
+    Binary(Vec<u8>),
+}
+
+pub trait Transformation: 'static {
+    fn apply(
+        &self,
+        params: &TransformationParams,
+        sources: &HashMap<String, Texture>,
```

Comment:
I think this is more self-explanatory and gives us compile time checks.
```suggestion
        sources: &HashMap<NodeID, Texture>,
```

Obviously, we need to define `pub struct NodeID(String)` as well.